### PR TITLE
chore: improve the stability of coverage test in CI

### PR DESCRIPTION
**Pull request by shawnh2**
1. improve test in `internal/provider/kubernetes/kubernetes_test.go` by adding empty check for `GatewayAPIResources` and nil check for `GatewayClassResources`. this unstable test discovered in serval PRs:
    - https://github.com/envoyproxy/gateway/actions/runs/8027649146/job/21931883980?pr=2684#step:4:250
    - https://github.com/envoyproxy/gateway/actions/runs/7978886579/job/21784989256#step:4:250
2. improve test in `internal/provider/kubernetes/routes_test.go` by adding resources check to prevent the httproute has been accessed before assigned. this unstable test discovered in serval PRs:
    - https://github.com/envoyproxy/gateway/actions/runs/7990660766/job/21819957769#step:4:950
    - https://github.com/envoyproxy/gateway/issues/2073

### CODE REVIEW

File: internal/provider/kubernetes/routes_test.go [hunk 1]
```diff
@@ -30,6 +31,11 @@ import (
 )
 
 func TestProcessHTTPRoutes(t *testing.T) {
+	const (
+		defaultWait = time.Second * 60
```
Comment (arkodg) [inline internal/provider/kubernetes/routes_test.go:35]: can we reduce this to 10s ?

Comment (shawnh2) [reply internal/provider/kubernetes/routes_test.go]: done

File: internal/provider/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -1547,6 +1620,9 @@ func TestNamespaceSelectorProvider(t *testing.T) {
 
 	require.Eventually(t, func() bool {
 		gatewayClassResources, ok := resources.GatewayAPIResources.Load(egv1a1.GatewayControllerName)
+		if gatewayClassResources == nil {
```
[the remark this answers is not in the document]
Comment (shawnh2) [reply internal/provider/kubernetes/kubernetes_test.go]: of course

**Review by cnvergence — APPROVED** [review]
lgtm

**Review by arkodg — APPROVED** [review]

**Pull request merged**